### PR TITLE
Introduce .encoding and .lang

### PR DESCRIPTION
**Pull request by kenchris**
***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Oct 17, 2019, 12:49 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Frawcdn.githack.com%2Fw3c%2Fweb-nfc%2Fa35eb3cce80d9ea9e06baff9946b1621bef1be52%2Findex.html%3FisPreview%3Dtrue)

```
Navigation Timeout Exceeded: 30000ms exceeded
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/web-nfc%23381.)._
</details>

### CODE REVIEW

**Comment by kenchris** [comment]
Not sure what is wrong with preview bot, but it works fine locally - no errors

File: index.html [hunk 1]
```diff
@@ -1678,6 +1684,30 @@ <h4>
...
+      The <dfn>encoding</dfn> attribute represents the
+      [=encoding/name|encoding name=] used for encoding the payload in the
+      case it is textual data.
+    </p>
+    <p>
+      The <dfn>lang</dfn> attribute represents the [=language tag=]
+      of the payload in the case that was encoded.
+    </p>
+    <p>
+      A <dfn>language tag</dfn> is a <a>string</a> that matches the
```
Comment (kenchris) [inline index.html:1697]: This is copy paste from our web app manifest spec

File: index.html [hunk 2]
```diff
@@ -1699,10 +1729,24 @@ <h4>
...
-      <dfn>recordType</dfn>, <dfn>mediaType</dfn>, <dfn>id</dfn>,
-      and <dfn>data</dfn>, respectively. The mapping from data types of an
+      with its <a>record type</a> <dfn>recordType</dfn>, and optional <a>record identifier</a>
+      <dfn>id</dfn> and payload data <dfn>data</dfn>.
+    </p>
+    <p data-dfn-for="NDEFRecordInit">
+      Additionally, there are additional optional fields that are only applicable
+      for certain <a>record types</a>:
+      <ul>
+        <li data-dfn-for="NDEFRecordInit">
```
Comment (kenchris) [inline index.html:1739]: Had to readd here (respec bug probably)

File: index.html [hunk 3]
```diff
@@ -2541,7 +2581,7 @@ <h3>The <dfn>NFCScanOptions</dfn> dictionary</h3>
       <pre class="idl">
         dictionary NFCScanOptions {
           USVString id = "";
-          NDEFRecordType recordType;
+          USVString recordType;
           USVString mediaType = "";
```
Comment (kenchris) [inline index.html:2585]: HTML spec uses DOMString for media type... guess we should do the same?

Things that include urls should be USVString I have been told

Comment (zolkis) [reply index.html:2585]: In Absolute URL the recordType would contain a URL

**Comment by zolkis** [comment]
Merge when you see it fit.

**Review by zolkis — APPROVED** [review]

File: index.html (hunk 3 above)
Comment (kenchris) [reply index.html:2585]: I thought we just set it to "absolute-url" and put the url in payload?

Comment (kenchris) [reply index.html:2585]: actually we should do that, we set it to "url" now

image

Comment (zolkis) [reply index.html:2585]: IMHO we should not deviate from NDEF semantics here. For TNF=3 let's use the TYPE field for the URL.

Comment (zolkis) [reply index.html:2585]: Maybe we can _also_ set `data` to the same, but `recordType` should contain the URL as well.

Comment (kenchris) [reply index.html:2585]: As we don't expose the TNF that will be confusing. How will people detect it is a url?

Comment (kenchris) [reply index.html:2585]: These are not supposed to contain payload so it's fine

Comment (zolkis) [reply index.html:2585]: Maybe we should expose TNF?

**Pull request merged**